### PR TITLE
Add `destroy` method

### LINKED ISSUE #1202: NetworkController API normalization: Add `destroy` to NetworkController
The extension version of NetworkController has a `destroy` method which [stops the block tracker if it is set](https://github.com/MetaMask/metamask-extension/blob/39f6042e6541e1badb9ee08098e55ce1f3c2dbef/app/scripts/controllers/network/network-controller.ts#L502). This assumes that the controller directly uses a network client that relies directly on `eth-json-rpc-middleware` instead of `web3-provider-engine`, which will be completed via https://github.com/metamask/core/issues/970. 

This is part of [a larger effort to normalize the API of both network controllers](https://github.com/MetaMask/controllers/issues/971), to make them easier to merge.

### PR DESCRIPTION
**Pull request by mikesposito**
## Description

This PR adds the `destroy` method to stop the block tracker polling.

## Changes

- **ADDED**: `destroy` method

## References

* Fixes #1202

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation for new or updated code as appropriate (note: this will usually be JSDoc)
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/network-controller/tests/NetworkController.test.ts [hunk 1]
```diff
@@ -5464,6 +5464,35 @@ describe('NetworkController', () => {
...
+    });
+
+    describe('if the blockTracker is undefined', () => {
+      it('should not throw errors', async () => {
+        await withController({}, async ({ controller }) => {
+          expect(async () => await controller.destroy()).not.toThrow();
+        });
+      });
+    });
+  });
```
Comment (mikesposito) [inline packages/network-controller/tests/NetworkController.test.ts:5495]: I'm not sure of which test cases we want to cover for this method, and what exactly we should observe / assert.
For now, it seemed logical to check if the `blockTracker.destroy` function has been called when `blockTracker` is defined, and that no error is thrown when it isn't.

File: packages/network-controller/src/NetworkController.ts [hunk 2]
```diff
@@ -791,6 +791,13 @@ export class NetworkController extends BaseControllerV2<
       await this.setActiveNetwork(specifier);
     }
   }
+
+  /**
+   * Stops the block tracker polling.
```
Comment (Gudahtt) [inline packages/network-controller/src/NetworkController.ts:811]: Nit: Here is the description used in the extension:

```suggestion
   * Deactivates the controller, stopping any ongoing polling.
   *
   * In-progress requests will not be aborted.
```

This seems a little more informative maybe? The clarification on in-progress requests is useful (we may want to find a way to abort those at some point, and change that behavior).

File: packages/network-controller/tests/NetworkController.test.ts (hunk 1 above)
Comment (Gudahtt) [reply packages/network-controller/tests/NetworkController.test.ts]: These look similar to the tests we have in the extension for this method as well! Agreed that these is what we want to cover.

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**